oauth2/cookies.go

```diff
@@ -35,47 +39,80 @@ func NewCookieJWT(secret string, duration time.Duration) Authenticator {
 }
 
 // Validate returns Principal of the Cookie if the Token is valid.
-func (c *cookie) Validate(ctx context.Context, r *http.Request) (Principal, error) {
+func (c *cookie) Validate(ctx context.Context, w http.ResponseWriter, r *http.Request) (Principal, error) {
 	cookie, err := r.Cookie(c.Name)
 	if err != nil {
 		return Principal{}, ErrAuthentication
 	}
-	return c.Tokens.ValidPrincipal(ctx, Token(cookie.Value), c.Duration)
+
+	p, err := c.Tokens.ValidPrincipal(ctx, Token(cookie.Value), c.Lifespan)
+	if err != nil {
+		return Principal{}, ErrAuthentication
+	}
+
+	// Refresh the token by extending its life another Inactivity duration
+	p, err = c.Tokens.ExtendPrincipal(ctx, p, c.Inactivity)
+	if err != nil {
+		return Principal{}, ErrAuthentication
+	}
+
+	token, err := c.Tokens.Create(ctx, p)
```

Comment:
Why do we need to Create a new token? Would be good to include a comment on this. @timraymond mentions that the token needs a new signature.